### ISSUE TITLE
capture and subprocess function looks not executing external code

### ISSUE THREAD

**Issue by handrake0724** [TACC/Lmod#588]
I am using Lmod 8.7.6 with lua 5.4.4 in Archlinux

In my module file, the following lines are used

```
local ncores=capture("expr `nproc` / 2")
local pinc=subprocess("mpicc -showme:compile")
```

those two lines should work correctly but it gives me nil value.

**Comment by rtmclay** [comment]
I am unable to reproduce your issue.  I created a command called nprocs which returns 4 and created the following module:
```
% module --raw show capture/1.0
----------------------------------------------------------------------------
   capture/1.0.lua:
----------------------------------------------------------------------------
local s = capture("expr `nprocs` / 2")
io.stderr:write("s:",s,"\n")
local s = subprocess("mpichversion")
io.stderr:write("s:",s,"\n")

% $LMOD_CMD bash load capture > /dev/null
s:2

s:MPICH Version:        4.0.2
MPICH Release date:     Thu Apr  7 12:34:45 CDT 2022
MPICH Device:           ch4:ofi
MPICH configure:        --prefix=/opt/apps/gcc-11/mpich/4.0.2 --enable-romio --disable-multi-aliases --enable-shared --enable-sharedlibs=gcc
MPICH CC:       gcc -g -O3 -fPIC   -O2
MPICH CXX:      g++ -g -O3 -fPIC  -O2
MPICH F77:      gfortran -g -O3 -fPIC -fallow-argument-mismatch  -O2
MPICH FC:       gfortran -g -O3 -fPIC -fallow-argument-mismatch  -O2
MPICH Custom Information:       

```
What happens for you when you run the same module?

**Comment by handrake0724** [comment]
I figured out what was the cause owing to your comments.
In summary, it was not because of the capture and subprocess issue but because of newline character.

Running the following script using $LMOD_CMD bash load command 
```
local ncores=capture("expr `nproc` / 2")
setenv("NPROC", ncores)
```

makes the following environment variable

```
s:32

LOADEDMODULES=capture/1.0;
NPROC=32
;
export NPROC;
```

NPROC variable has a newline character between 32 and ;.
So `module load capture` command gives me the following message
   bash: syntax error near unexpected token `;'

I could fix the issue by modifying as follows:

```
local ncores=capture("expr `nproc` / 2"):gsub("[\n\r]", "")
setenv("NPROC", ncores)
```

But it is a little weird for me since I have used that script for some years without any problem, though.
